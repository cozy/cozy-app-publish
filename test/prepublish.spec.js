/* eslint-env jest */
const prepublishLib = require('../lib/prepublish')

describe('Prepublish script', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  const optionsMock = {
    appBuildUrl: 'http://example.com',
    appSlug: 'cozy-mock',
    appType: 'webapp',
    appVersion: '0.0.1',
    registryUrl: 'http://registry.mock/',
    registryEditor: 'Cozy',
    registryToken: 'a5464f54e654c6546b54a56a'
  }

  it('generates sha256', async () => {
    await expect(prepublishLib(optionsMock)).resolves.toMatchSnapshot()
  })

  it('handles prepublish hook', async () => {
    const options = { ...optionsMock, prepublishHook: './test/__mocks__/prepublish-hook' }
    await expect(prepublishLib(options)).resolves.toMatchSnapshot()
  })

  it('ignores invalid prepublish hook path', async () => {
    const options = { ...optionsMock, prepublishHook: './test/__mocks__/not-existing-hook' }
    await expect(prepublishLib(options)).resolves.toMatchSnapshot()
  })

  it('ignores prepublish hook thrown error', async () => {
    const options = { ...optionsMock, prepublishHook: './test/__mocks__/prepublish-errored-hook' }
    await expect(prepublishLib(options)).resolves.toMatchSnapshot()
  })

  it('sanitize options from hook script', async () => {
    const options = { ...optionsMock, prepublishHook: './test/__mocks__/prepublish-unsanitized-hook' }
    await expect(prepublishLib(options)).resolves.toMatchSnapshot()
  })

  it('check for undefined mandatory options', async () => {
    const options = { ...optionsMock, prepublishHook: './test/__mocks__/prepublish-missing-options-hook' }
    await expect(prepublishLib(options)).rejects.toThrowErrorMatchingSnapshot()
  })
})
