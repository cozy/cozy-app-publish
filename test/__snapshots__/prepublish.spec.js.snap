// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prepublish script check for undefined mandatory options 1`] = `"appSlug cannot be undefined"`;

exports[`Prepublish script generates sha256 1`] = `
Object {
  "appBuildUrl": "http://example.com",
  "appSlug": "cozy-mock",
  "appType": "webapp",
  "appVersion": "0.0.1",
  "buildCommit": undefined,
  "registryEditor": "Cozy",
  "registryToken": "a5464f54e654c6546b54a56a",
  "registryUrl": "http://registry.mock/",
  "sha256Sum": "3587cb776ce0e4e8237f215800b7dffba0f25865cb84550e87ea8bbac838c423",
  "spaceName": undefined,
}
`;

exports[`Prepublish script handles prepublish hook 1`] = `
Object {
  "appBuildUrl": "http://cozy.io",
  "appSlug": "cozy-mock",
  "appType": "webapp",
  "appVersion": "0.0.1",
  "buildCommit": undefined,
  "registryEditor": "Cozy",
  "registryToken": "a5464f54e654c6546b54a56a",
  "registryUrl": "http://registry.mock/",
  "sha256Sum": "67ebd36d450fe0382f97f296c1ee23d80463bd35dd9618a398307ad6509ba59d",
  "spaceName": undefined,
}
`;

exports[`Prepublish script ignores invalid prepublish hook path 1`] = `
Object {
  "appBuildUrl": "http://example.com",
  "appSlug": "cozy-mock",
  "appType": "webapp",
  "appVersion": "0.0.1",
  "buildCommit": undefined,
  "registryEditor": "Cozy",
  "registryToken": "a5464f54e654c6546b54a56a",
  "registryUrl": "http://registry.mock/",
  "sha256Sum": "3587cb776ce0e4e8237f215800b7dffba0f25865cb84550e87ea8bbac838c423",
  "spaceName": undefined,
}
`;

exports[`Prepublish script ignores prepublish hook thrown error 1`] = `
Object {
  "appBuildUrl": "http://example.com",
  "appSlug": "cozy-mock",
  "appType": "webapp",
  "appVersion": "0.0.1",
  "buildCommit": undefined,
  "registryEditor": "Cozy",
  "registryToken": "a5464f54e654c6546b54a56a",
  "registryUrl": "http://registry.mock/",
  "sha256Sum": "3587cb776ce0e4e8237f215800b7dffba0f25865cb84550e87ea8bbac838c423",
  "spaceName": undefined,
}
`;

exports[`Prepublish script sanitize options from hook script 1`] = `
Object {
  "appBuildUrl": "http://example.com",
  "appSlug": "cozy-mock",
  "appType": "webapp",
  "appVersion": "0.0.1",
  "buildCommit": undefined,
  "registryEditor": "Cozy",
  "registryToken": "a5464f54e654c6546b54a56a",
  "registryUrl": "http://registry.mock/",
  "sha256Sum": "3587cb776ce0e4e8237f215800b7dffba0f25865cb84550e87ea8bbac838c423",
  "spaceName": undefined,
}
`;
